packages/flutter_tools/lib/src/android/android_builder.dart

```diff
@@ -54,23 +53,30 @@ class _AndroidBuilderImpl extends AndroidBuilder {
   @override
   Future<void> buildAar({
     @required FlutterProject project,
-    @required AndroidBuildInfo androidBuildInfo,
-    @required String target,
-    @required String outputDir,
+    @required Set<AndroidBuildInfo> androidBuildInfo,
+    @required String outputDirectoryPath,
   }) async {
     try {
       Directory outputDirectory =
-        fs.directory(outputDir ?? project.android.buildDirectory);
+        fs.directory(outputDirectoryPath ?? project.android.buildDirectory);
       if (project.isModule) {
         // Module projects artifacts are located in `build/host`.
         outputDirectory = outputDirectory.childDirectory('host');
       }
-      await buildGradleAar(
-        project: project,
-        androidBuildInfo: androidBuildInfo,
-        target: target,
-        outputDir: outputDirectory,
-        printHowToConsumeAaar: true,
+      for (AndroidBuildInfo androidBuildInfo in androidBuildInfo) {
+        await buildGradleAar(
+          project: project,
+          androidBuildInfo: androidBuildInfo,
+          outputDirectory: outputDirectory,
+        );
+      }
+      printHowToConsumeAar(
+        buildModes: androidBuildInfo
+          .map<String>((AndroidBuildInfo androidBuildInfo) {
+            return androidBuildInfo.buildInfo.modeName;
+          }).toSet(),
```

Comment:
should we order them? Since we tend to keep debug/profile/release in that order in many other docs etc